Add `recordDiff` to ConfigPolicy and generator (#6095)
open-cluster-management/rhacm-docs: governance/policy_ctrl_adv_config.adoc

  installNamespace: open-cluster-management-agent-addon
----

> Additionally, for each `spec.object-template[]` in a `ConfigurationPolicy`, you can set the parameter `recordDiff` to "Log". The difference between the `objectDefinition` and the object on the managed cluster is logged in the `config-policy-controller` pod on the managed cluster. View the following example:

Reviewer: Is this info apart of the YAML: 

Logging the diff:
--- default/my-configmap : existing
+++ default/my-configmap : updated
@@ -2,3 +2,3 @@

```suggestion
Additionally, for each `spec.object-template[]` in a `ConfigurationPolicy` resource, you can set the parameter `recordDiff` to `Log`. The difference between the `objectDefinition` and the object on the managed cluster is logged in the `config-policy-controller` pod on the managed cluster. View the following example:
```

Author: > Is this info apart of the YAML:
> 
> Logging the diff: --- default/my-configmap : existing +++ default/my-configmap : updated @@ -2,3 +2,3 @@

No, not part of the YAML--that's what the logs look like in the `config-policy-controller` Pod when the diff is logged.

Author: Oh actually, I see what you mean. It'd probably be clearer to have a YAML example and then the log example. I'll add one.

[source,yaml]
----
Logging the diff:
--- default/my-configmap : existing
+++ default/my-configmap : updated
@@ -2,3 +2,3 @@
 data:
-  fieldToUpdate: "1"
+  fieldToUpdate: "2"
 kind: ConfigMap
----

> *NOTE:* Use caution to not log the difference for a secure object. The difference is logged in plain text.

Reviewer: I changed this to important, so that users understand the severity.
```suggestion
*Important:* Avoid logging the difference for a secure object. The difference is logged in plain text.
```

  installNamespace: open-cluster-management-agent-addon
----

Additionally, for each `spec.object-template[]` in a `ConfigurationPolicy` resource, you can set the parameter `recordDiff` to `Log`. The difference between the `objectDefinition` and the object on the managed cluster is logged in the `config-policy-controller` pod on the managed cluster. View the following example:

This `ConfigurationPolicy` resource with `recordDiff: Log`:

[source,yaml]
----
apiVersion: policy.open-cluster-management.io/v1
kind: ConfigurationPolicy
metadata:
  name: my-config-policy
spec:
  object-templates:
  - complianceType: musthave
    recordDiff: Log
    objectDefinition:
      apiVersion: v1
      kind: ConfigMap
      metadata:
        name: my-configmap
      data:
        fieldToUpdate: "2"
----

> If the `ConfigMap` resource on the cluster had `fieldToUpdate: "1"`, then the diff would appear in the `config-policy-controller` Pod logs as:

Reviewer: ```suggestion
If the `ConfigMap` resource on the cluster lists `fieldToUpdate: "1"`, then the diff appears in the `config-policy-controller` pod logs with the following information:
```

  installNamespace: open-cluster-management-agent-addon
----

Additionally, for each `spec.object-template[]` in a `ConfigurationPolicy` resource, you can set the parameter `recordDiff` to `Log`. The difference between the `objectDefinition` and the object on the managed cluster is logged in the `config-policy-controller` pod on the managed cluster. View the following example:

This `ConfigurationPolicy` resource with `recordDiff: Log`:

[source,yaml]
----
apiVersion: policy.open-cluster-management.io/v1
kind: ConfigurationPolicy
metadata:
  name: my-config-policy
spec:
  object-templates:
  - complianceType: musthave
    recordDiff: Log
    objectDefinition:
      apiVersion: v1
      kind: ConfigMap
      metadata:
        name: my-configmap
      data:
        fieldToUpdate: "2"
----

If the `ConfigMap` resource on the cluster had `fieldToUpdate: "1"`, then the diff would appear in the `config-policy-controller` Pod logs as:

> [source]

Reviewer: the source syntax is not needed here I think. Or maybe we use bash? Is it accurate to say this is bash syntax @dhaiducek 
```suggestion
[source,bash]
```

Author: Ah, I didn't realize that wasn't necessary. It's not really bash, so I'll just remove it. Thanks!
